src/main/java/frc/robot/drive/SpeedControllerFactory.java

```diff
@@ -0,0 +1,20 @@
+package frc.robot.drive;
+
+import com.revrobotics.CANSparkMaxLowLevel;
+
+public class SpeedControllerFactory {
+    public static SpeedControllerEncoder create(int motorID, MotorType type) {
+        switch (type) {
+            case TALON_SRX:
+                return new TalonController(motorID);
+
+            case SPARK_MAX_BRUSHED:
+                return new SparkMaxController(motorID, CANSparkMaxLowLevel.MotorType.kBrushed);
+
+            case SPARK_MAX_BRUSHLESS:
+                return new SparkMaxController(motorID, CANSparkMaxLowLevel.MotorType.kBrushless);
+        }
+
+        return null;
```

Comment:
I wonder if this should throw an IllegalArgumentException in this case, so we know right away that something bad is happening. Otherwise you don't find out about the problem until you dereference the null (NullPointerException).